monai/apps/auto3dseg/algorithms/templates/dints/scripts/train.py

```diff
@@ -0,0 +1,400 @@
+
+                    val_outputs = post_pred(val_outputs[0, ...])
+                    val_outputs = val_outputs[None, ...]
+
+                    if softmax:
+                        val_labels = post_label(val_labels[0, ...])
+                        val_labels = val_labels[None, ...]
+
+                    value = compute_meandice(y_pred=val_outputs, y=val_labels, include_background=not softmax)
+
+                    print(_index + 1, "/", len(val_loader), value)
+
+                    metric_count += len(value)
+                    metric_sum += value.sum().item()
+                    metric_vals = value.cpu().numpy()
+                    if len(metric_mat) == 0:
+                        metric_mat = metric_vals
+                    else:
+                        metric_mat = np.concatenate((metric_mat, metric_vals), axis=0)
+
+                    for _c in range(metric_dim):
+                        val0 = torch.nan_to_num(value[0, _c], nan=0.0)
+                        val1 = 1.0 - torch.isnan(value[0, 0]).float()
+                        metric[2 * _c] += val0 * val1
+                        metric[2 * _c + 1] += val1
+
+                    _index += 1
+
+                if torch.cuda.device_count() > 1:
+                    dist.barrier()
+                    dist.all_reduce(metric, op=torch.distributed.ReduceOp.SUM)
+
+                metric = metric.tolist()
+                if torch.cuda.device_count() == 1 or dist.get_rank() == 0:
+                    for _c in range(metric_dim):
+                        print(f"evaluation metric - class {_c + 1:d}:", metric[2 * _c] / metric[2 * _c + 1])
+                    avg_metric = 0
+                    for _c in range(metric_dim):
+                        avg_metric += metric[2 * _c] / metric[2 * _c + 1]
+                    avg_metric = avg_metric / float(metric_dim)
+                    print("avg_metric", avg_metric)
+
+                    writer.add_scalar("Accuracy/validation", avg_metric, epoch)
+
+                    if avg_metric > best_metric:
+                        best_metric = avg_metric
+                        best_metric_epoch = epoch + 1
+                        if torch.cuda.device_count() > 1:
+                            torch.save(model.module.state_dict(), os.path.join(ckpt_path, "best_metric_model.pt"))
+                        else:
+                            torch.save(model.state_dict(), os.path.join(ckpt_path, "best_metric_model.pt"))
+                        print("saved new best metric model")
+
+                        dict_file = {}
+                        dict_file["best_avg_dice_score"] = float(best_metric)
+                        dict_file["best_avg_dice_score_epoch"] = int(best_metric_epoch)
+                        dict_file["best_avg_dice_score_iteration"] = int(idx_iter)
+                        with open(os.path.join(ckpt_path, "progress.yaml"), "a") as out_file:
+                            yaml.dump([dict_file], stream=out_file)
```

Comment:
The yaml.dump will append the best_metric. If that's not intended, the following works well for me.

```python
ConfigParser.export_config_file(dict_file, os.path.join(ckpt_path, "progress.yaml"))
```

Reply:
It is intended to know the history.